peerconnection_renegotiation_test.go

```diff
@@ -899,3 +899,87 @@ func TestNegotiationNeededRemoveTrack(t *testing.T) {
 	assert.NoError(t, pcOffer.Close())
 	assert.NoError(t, pcAnswer.Close())
 }
+
+func TestNegotiationNeededStress(t *testing.T) {
+	api := NewAPI()
+	lim := test.TimeOut(time.Second * 30)
+	defer lim.Stop()
+
+	report := test.CheckRoutines(t)
+	defer report()
+
+	api.mediaEngine.RegisterDefaultCodecs()
+	pcA, pcB, err := api.newPair(Configuration{})
+	assert.NoError(t, err)
+	defer pcA.Close()
+	defer pcB.Close()
+
+	var mu sync.Mutex
+	pcA.OnNegotiationNeeded(func() {
+		mu.Lock()
+		assert.NoError(t, signalPair(pcA, pcB))
+		mu.Unlock()
+	})
+
+	pcB.OnNegotiationNeeded(func() {
+		mu.Lock()
+		assert.NoError(t, signalPair(pcB, pcA))
+		mu.Unlock()
+	})
+
+	done := make(chan bool)
+	go func() {
+		for {
+			time.Sleep(time.Duration(rand.Intn(100))*time.Millisecond + 10)
```

Comment:
you can expose more errors by reducing the `10` and `50` lower bounds here and below